commons/com.b2international.index/src/com/b2international/index/revision/StagingArea.java

```diff
@@ -696,7 +696,10 @@ public StagingArea stageNew(String key, Object newDocument, boolean commit) {
 		if (stagedObjects.containsKey(objectId)) {
 			StagedObject currentStagedObject = stagedObjects.get(objectId);
 			if (!currentStagedObject.isCommit() && currentStagedObject.getObject() instanceof Revision && newDocument instanceof Revision) {
-				stagedObjects.put(objectId, changed(newDocument, new RevisionDiff((Revision) currentStagedObject.getObject(), (Revision) newDocument), commit));
+				RevisionDiff diff = new RevisionDiff((Revision) currentStagedObject.getObject(), (Revision) newDocument);
+				if (diff.hasChanges()) {
+					stagedObjects.put(objectId, changed(newDocument, diff, commit));
```

Comment:
Additional tests were requested on the sibling PR #1281, please add that test case to this branch as well.